vendor/k8s.io/kubernetes/test/e2e/framework/test_context.go

```diff
@@ -227,7 +227,7 @@ func RegisterCommonFlags() {
 	flag.StringVar(&TestContext.ReportPrefix, "report-prefix", "", "Optional prefix for JUnit XML reports. Default is empty, which doesn't prepend anything to the default name.")
 	flag.StringVar(&TestContext.ReportDir, "report-dir", "", "Path to the directory where the JUnit XML reports should be saved. Default is empty, which doesn't generate these reports.")
 	flag.Var(utilflag.NewMapStringBool(&TestContext.FeatureGates), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features.")
-	flag.StringVar(&TestContext.Viper, "viper-config", "e2e", "The name of the viper config i.e. 'e2e' will read values from 'e2e.json' locally.  All e2e parameters are meant to be configurable by viper.")
+	flag.StringVar(&TestContext.Viper, "viper-config", os.Getenv("VIPERCONFIG"), "The name of the viper config i.e. 'e2e' will read values from 'e2e.json' locally.  All e2e parameters are meant to be configurable by viper.")
```

Comment:
You can set this a different way.  in `openshift-tests run-test` code read os.Getenv("VIPERCONFIG") and set it on exutil.TestContext.Viper

Reply:
Thanks Clayton, is the new location inside `initProvider` okay?